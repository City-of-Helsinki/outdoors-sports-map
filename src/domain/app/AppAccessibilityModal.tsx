import Button from "react-bootstrap/Button";
import Modal from "react-bootstrap/Modal";
import { useTranslation } from "react-i18next";

import AccessibilityStatementEn from "./accessibilityStatement/AccessibilityStatementEn";
import AccessibilityStatementFi from "./accessibilityStatement/AccessibilityStatementFi";
import AccessibilityStatementSv from "./accessibilityStatement/AccessibilityStatementSv";

type AccessibilityStatementLanguage = "fi" | "sv" | "en";

const modalContentMap = {
  fi: AccessibilityStatementFi,
  sv: AccessibilityStatementSv,
  en: AccessibilityStatementEn,
} as const;

type Props = {
  onClose: () => void;
  show: boolean;
};

function AppAccessibilityModal({ onClose, show }: Props) {
  const {
    t,
    i18n: { language },
  } = useTranslation();

  const Content = modalContentMap[
    language as AccessibilityStatementLanguage
  ] ?? (
    <span lang="en">
      Could not find accessibility statement for selected language
    </span>
  );

  return (
    <Modal show={show} onHide={onClose} size="lg" animation={false}>
      <Modal.Header closeButton closeLabel={t("APP.MODAL.CLOSE")}>
        <Modal.Title>
          <h2>{t("APP.INFO_MENU.ACCESSIBILITY")}</h2>
        </Modal.Title>
      </Modal.Header>
      <Modal.Body>
        <Content />
      </Modal.Body>
      <Modal.Footer>
        {/* Give screen reader users a control for closing the modal after */}
        {/* they've gone through the content. */}
        <Button variant="primary" onClick={onClose}>
          {t("APP.MODAL.CLOSE")}
        </Button>
      </Modal.Footer>
    </Modal>
  );
}

export default AppAccessibilityModal;
