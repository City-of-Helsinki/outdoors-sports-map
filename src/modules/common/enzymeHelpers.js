import React, { } from 'react';
// eslint-disable-next-line import/no-extraneous-dependencies
import { mount as enzymeMount } from 'enzyme';
import { Provider } from 'react-redux';

import createStore from '../../bootstrap/createStore';
import TranslationProvider from './components/translation/TranslationProvider';

// eslint-disable-next-line import/prefer-default-export
export const mount = async (element) => {
  const store = await createStore();

  return enzymeMount(
    <Provider store={store}>
      <TranslationProvider>
        {element}
      </TranslationProvider>
    </Provider>,
  );
};
