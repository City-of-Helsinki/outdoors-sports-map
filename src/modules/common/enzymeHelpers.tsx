import { mount as enzymeMount, MountRendererProps } from "enzyme";
import { ReactNode } from "react";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";

import createStore from "../../bootstrap/createStore";
import TranslationProvider from "./components/translation/TranslationProvider";

export const mount = (element: ReactNode, options?: MountRendererProps) => {
  const { store } = createStore();

  return enzymeMount(
    <Provider store={store}>
      <TranslationProvider>
        <MemoryRouter>{element}</MemoryRouter>
      </TranslationProvider>
    </Provider>,
    options
  );
};
