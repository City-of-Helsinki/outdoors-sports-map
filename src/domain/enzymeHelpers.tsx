import { mount as enzymeMount, MountRendererProps } from "enzyme";
import get from "lodash/get";
import omit from "lodash/omit";
import { ReactNode } from "react";
import { HelmetProvider } from "react-helmet-async";
import { Provider } from "react-redux";
import { MemoryRouter } from "react-router-dom";

import { AppState } from "./app/appConstants";
import createStore from "./bootstrap/createStore";
import TranslationProvider from "./i18n/I18nTranslationProvider";

type Options = MountRendererProps & { preloadedState: Partial<AppState> };

export const mount = (element: ReactNode, options?: Options) => {
  const enzymeOptions = omit(options, "preloadedState");
  const preloadedState = get(options, "preloadedState");
  // @ts-ignore
  const { store } = createStore(preloadedState);

  return enzymeMount(
    <Provider store={store}>
      <TranslationProvider>
        <MemoryRouter>
          <HelmetProvider>{element}</HelmetProvider>
        </MemoryRouter>
      </TranslationProvider>
    </Provider>,
    enzymeOptions
  );
};
