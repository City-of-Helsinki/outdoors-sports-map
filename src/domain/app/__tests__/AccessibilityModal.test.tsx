import { mount } from "../../enzymeHelpers";
import AppAccessibilityModal from "../AppAccessibilityModal";

const getWrapper = () =>
  mount(
    <AppAccessibilityModal
      onClose={() => {
        // pass
      }}
      show
    />
  );

describe("<AppAccessibilityModal />", () => {
  it("renders correctly", () => {
    const wrapper = getWrapper();

    expect(wrapper.html()).toMatchSnapshot();
  });
});
