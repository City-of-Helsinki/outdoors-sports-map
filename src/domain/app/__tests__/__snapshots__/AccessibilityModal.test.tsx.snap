// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<AppAccessibilityModal /> renders correctly 1`] = `"<div class=\\"modal-backdrop show\\"></div><div role=\\"dialog\\" aria-modal=\\"true\\" style=\\"display: block;\\" class=\\"modal\\" tabindex=\\"-1\\"><div class=\\"modal-dialog modal-lg\\"><div class=\\"modal-content\\"><div class=\\"modal-header\\"><div class=\\"modal-title h4\\"><h2>Saavutettavuusseloste</h2></div><button type=\\"button\\" class=\\"close\\"><span aria-hidden=\\"true\\">×</span><span class=\\"sr-only\\">Sulje</span></button></div><div class=\\"modal-body\\"><p>Tämä saavutettavuusseloste koskee Helsingin kaupungin verkkosivustoa ulkoliikunta.fi.</p><h3>Sivustoa koskevat lain säädökset</h3><p>Sivusto on julkaistu aiemmin kuin 23.9.2018. Sivuston tulee täyttää lain edellyttämät saavutettavuuden vaatimukset 23.9.2020 päättyvän siirtymäajan jälkeen.</p><h3>Kaupungin tavoite</h3><p>Digitaalisten palveluiden saavutettavuudessa Helsingin tavoitteena on pyrkiä vähintään WCAG ohjeiston mukaiseen AA- tai sitä parempaan tasoon, mikäli se on kohtuudella mahdollista.</p><h3>Vaatimustenmukaisuustilanne</h3><p>Tämä verkkosivusto täyttää lain asettamat kriittiset saavutettavuusvaatimukset WCAG v2.1 -tason AA mukaisesti.</p><h3>Saavutettavuusselosteen laatiminen</h3><p>Tämä seloste on laadittu&nbsp;13.8.2021.</p><h3>Saavutettavuuden arviointi</h3><p>Saavutettavuuden arvioinnissa on noudatettu Helsingin kaupungin työohjetta ja menetelmiä, jotka pyrkivät varmistamaan sivuston saavutettavuuden kaikissa työvaiheissa.</p><p>Saavutettavuus on tarkistettu ulkopuolisen asiantuntijan suorittamana auditointina sekä itsearviona. Ulkopuolisen asiantuntija-auditoinnin on tehnyt Palvelukeskus Helsinki.</p><p>Saavutettavuus on tarkistettu käyttäen ohjelmallista saavutettavuustarkistusta sekä sivuston manuaalista tarkistusta. Testauksessa syntyneet toiminnalliset havainnot pätevät sekä työpöytä- että mobiililaiteympäristöön.</p><p>Manuaalisessa testauksessa on käytetty Chrome- ja Firefox-selaimia, niiden 200%-tiloja sekä tietoteknisiä apuvälineitä, kuten ruudunlukuohjelmia, ohjaimia ja erikoisnäppäimistöjä. Mobiilitestaus toteutettiin iOS- ja Android-käyttöjärjestelmillä ja niille tarkoitetuilla ruudunlukuohjelmilla.</p><p>Arviointityökalujen raportoimat sekä auditoinnin epäkohdat on tarkastettu ja tarvittaessa korjattu.</p><h3>Saavutettavuusselosteen päivittäminen</h3><p>Sivuston saavutettavuudesta huolehditaan jatkuvalla valvonnalla tekniikan tai sisällön muuttuessa, sekä määräajoin, kuitenkin vähintään kerran vuodessa, suoritettavalla tarkistuksella. Tätä selostetta päivitetään sivuston muutosten ja saavutettavuuden tarkistusten yhteydessä.</p><h3>Palaute ja yhteystiedot</h3><p>Sivuston saavutettavuudesta vastaa kulttuurin ja vapaa-ajan toimialan viestintä, puhelinvaihde <a href=\\"tel:093101060\\">09 310 1060</a>.</p><h3>Ilmoittaminen ei-saavutettavasta sisällöstä</h3><p>Jos käyttäjä kokee, etteivät saavutettavuuden vaatimukset kuitenkaan täyty, voi tästä tehdä ilmoituksen palautelomakkeella <a href=\\"https://palautteet.hel.fi\\">palautteet.hel.fi</a>.</p><p><a href=\\"https://www.hel.fi/helsinki/fi/kaupunki-ja-hallinto/osallistu-ja-vaikuta/palaute\\">Siirry palautelomakkeeseen</a></p><h3>Tietojen pyytäminen saavutettavassa muodossa</h3><p>Jos käyttäjä ei koe saavansa sivuston sisältöä saavutettavassa muodossa,&nbsp;voi käyttäjä pyytää näitä tietoja palautelomakkeella&nbsp;<a href=\\"https://palautteet.hel.fi\\">palautteet.hel.fi</a>.&nbsp;Tiedusteluun pyritään vastaamaan kohtuullisessa ajassa.</p><p><a href=\\"https://www.hel.fi/helsinki/fi/kaupunki-ja-hallinto/osallistu-ja-vaikuta/palaute\\">Siirry palautelomakkeeseen</a></p><h3>Saavutettavuuden oikeussuoja, Täytäntöönpanomenettely</h3><p>Mikäli henkilö kokee, ettei hänen ilmoitukseensa tai tiedusteluunsa ole vastattu tai vastaus ei ole tyydyttävä, voi asiasta tehdä ilmoituksen Etelä-Suomen aluehallintovirastoon. Etelä-Suomen aluehallintoviraston sivulla kerrotaan tarkasti, miten asia käsitellään.</p><p>Etelä-Suomen aluehallintovirasto</p><p>Saavutettavuuden valvonnan yksikkö</p><p><a href=\\"https://www.saavutettavuusvaatimukset.fi\\">www.saavutettavuusvaatimukset.fi</a></p><p><a href=\\"mailto:saavutettavuus@avi.fi\\">saavutettavuus@avi.fi</a></p><p>Puhelinvaihde: <a href=\\"tel:0295016000\\">0295 016 000</a></p><p>Avoinna: ma–pe klo 8.00–16.15</p>&nbsp;<h3>Helsingin kaupunki ja saavutettavuus</h3><p>Helsingin kaupungin tavoitteena on olla kaikille esteetön ja saavutettava kaupunki. Kaupungin tavoitteena on, että Helsingissä on kaikkien kaupunkilaisten mahdollisimman helppo liikkua ja&nbsp;toimia, ja että&nbsp;kaikki sisältö ja palvelut olisivat kaikkien saavutettavissa.</p><p>Kaupunki edistää digitaalisten palveluiden saavutettavuutta yhdenmukaistamalla julkaisutyötä ja järjestämällä saavutettavuuteen keskittyvää koulutusta henkilökunnalleen.</p><p>Sivustojen saavutettavuuden tasoa&nbsp;seurataan jatkuvasti sivustoja ylläpidettäessä. Havaittuihin puutteisiin&nbsp;reagoidaan&nbsp;välittömästi.&nbsp;Tarvittavat&nbsp;muutokset&nbsp;pyritään suorittamaan mahdollisimman nopeasti.</p><h3>Vammaiset ja avustavien teknologioiden käyttäjät</h3><p>Kaupunki tarjoaa neuvontaa ja tukea vammaisille ja avustavien teknologioiden käyttäjille. Tukea on saatavilla kaupungin sivuilla ilmoitetuilta neuvontasivuilta sekä puhelinneuvonnasta.</p><p><a href=\\"https://www.hel.fi/helsinki/fi/kaupunki-ja-hallinto/hallinto/viestinta/viestinta-ja-neuvonta\\">Siirry kaupungin neuvontasivuille</a></p><h3>Saavutettavuusselosteen hyväksyntä</h3><p>Tämän selosteen on hyväksynyt&nbsp;1.10.2021<br>Kulttuurin ja vapaa-ajan toimiala<br>Helsingin kaupunki<br></p></div><div class=\\"modal-footer\\"><button type=\\"button\\" class=\\"btn btn-primary\\">Sulje</button></div></div></div></div>"`;
